presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/query/PinotQueryGeneratorContext.java

```diff
@@ -553,6 +558,26 @@ public PinotQueryGeneratorContext withVariablesInAggregation(Set<VariableReferen
                 useSqlSyntax);
     }
 
+    public PinotQueryGeneratorContext withDistinctLimit(LinkedHashSet<VariableReferenceExpression> newGroupByColumns, long limit)
+    {
+        int intLimit = checkForValidLimit(limit);
+        checkSupported(!hasLimit(), "Limit already exists. Pinot doesn't support limit on top of another limit");
+        checkSupported(!hasGroupBy(), "GroupBy already exists. Pinot doesn't support Distinct on top of another Group By");
```

Comment:
nit: Should we omit adding a helper method when it is used only once ? I think this can be inlined ?

Reply:
there are many other places uses this, I replaces all of them using `hasGroupBy()`.